src/lib.rs

```diff
@@ -394,35 +398,98 @@ impl Check {
                                 Some(version),
                             )?;
 
-                            Ok(run_check_release(
-                                &mut config,
-                                crate_name,
-                                current_crate,
-                                baseline_crate,
-                                self.release_type,
-                            )?)
+                            Ok((
+                                crate_name.clone(),
+                                Some(run_check_release(
+                                    &mut config,
+                                    crate_name,
+                                    current_crate,
+                                    baseline_crate,
+                                    self.release_type,
+                                )?),
+                            ))
                         }
                     })
                     .collect()
             }
         };
-        let success = all_outcomes
-            .into_iter()
-            .fold_ok(true, std::ops::BitAnd::bitand)?;
+        let crate_reports: BTreeMap<String, CrateReport> = {
+            let mut reports = BTreeMap::new();
+            for outcome in all_outcomes {
+                let (name, outcome) = outcome?;
+                if let Some(outcome) = outcome {
+                    reports.insert(name, outcome);
+                }
+            }
+            reports
+        };
 
-        Ok(Report { success })
+        Ok(Report { crate_reports })
     }
 }
 
+/// Report of semver check of one crate.
+#[non_exhaustive]
+#[derive(Debug)]
+pub struct CrateReport {
+    /// Bump between the current version and the baseline one.
+    detected_bump: ActualSemverUpdate,
+    /// Minimum bump required to respect semver.
+    /// For example, if the crate contains breaking changes, this is [`ReleaseType::Major`].
+    /// If no bump is required, this is [`Option::None`].
+    required_bump: Option<ReleaseType>,
+}
+
+impl CrateReport {
+    /// Check if the semver check was successful.
+    /// `true` if required bump <= detected bump.
+    pub fn success(&self) -> bool {
+        match self.required_bump {
+            None => true,
+            Some(required_bump) => {
+                match self.detected_bump {
+                    // If user bumped the major version, any breaking change is accepted.
+                    ActualSemverUpdate::Major => true,
+                    ActualSemverUpdate::Minor => {
+                        matches!(required_bump, ReleaseType::Minor | ReleaseType::Patch)
+                    }
+                    ActualSemverUpdate::Patch | ActualSemverUpdate::NotChanged => {
+                        required_bump == ReleaseType::Patch
+                    }
```

Comment:
Does this need updating for the changes to the semantics of the struct's fields? It seems to me like some of these could be `assert` instead of comparisons.

Reply:
Yes, you are right. I think it should never happen that the update of `detected_bump` is the same of `required_bump`, right?
Maybe we can add some warnings in this case. Or do you prefer to write an assert? Sounds scary 😅

Reply:
Let's assert. If our design behaves unexpectedly, it should fail loudly and quickly so we can figure out the failure mode and fix it.

I think `required_bump` should never be equal to or lesser than `detected_bump` in this design. If we're wrong about that, there's some serious issue we've overlooked and the outcome shouldn't be trusted anyway.

Reply:
I added the assert in [9cf72ee](https://github.com/obi1kenobi/cargo-semver-checks/pull/398/commits/9cf72ee870d0545448cc95e9806bc591c7b41813)